static/js/stream_data.js

```diff
@@ -268,7 +262,9 @@ exports.update_calculated_fields = function (sub) {
                                  !sub.invite_only;
     sub.preview_url = hash_util.by_stream_uri(sub.stream_id);
     sub.can_add_subscribers = !page_params.is_guest && (!sub.invite_only || sub.subscribed);
-    exports.render_stream_description(sub);
+    if (sub.rendered_description !== undefined) {
+        sub.rendered_description = sub.rendered_description.replace('<p>', '').replace('</p>', '');
+    }
```

Comment:
Why do we need to remove the `p` type tags here?  I'm a little skeptical that doing so is correct (though I see we had it in the previous implementation).

Reply:
Without stripping the `<p>, </p>` tags, the edit symbol/button would get moved to the next line.
Like so:
![](https://user-images.githubusercontent.com/29123352/52519168-5af43a00-2c7d-11e9-8976-a589a28e9c91.png)
Which is kinda ugly.

Reply:
I see.  I think that's fine, what I don't like about this model is that we strip the `<p>` tags every time we call this function, regardless of whether the `rendered_description` has actually changed.  It's definitely a code smell, even if it happens to be correct.  

I think we could potentially address this in other ways, though -- e.g. either making the UI layer insert the edit widget inside the `<p>` tag, or tweaking the CSS.  @synicalsyntax do you see a natural way to avoid that line break with CSS without this line?